server/swap/swap.go

```diff
@@ -2021,10 +2083,27 @@ func (s *Swapper) handleInit(user account.AccountID, msg *msgjson.Message) *msgj
 
 	s.setLiveWaiter(user, msg)
 
+	// Do not search for the transaction past the inaction deadline. For maker,
+	// this is bTimeout after match request. For taker, this is bTimeout after
+	// maker's swap reached swapConfs.
+	lastEvent := stepInfo.match.time // NewlyMatched - the match request time, not matchTime
+	if stepInfo.step == order.MakerSwapCast {
+		lastEvent = stepInfo.match.makerStatus.swapConfTime()
+	}
+	expireTime := time.Now().Add(txWaitExpiration).UTC()
+	if lastEvent.IsZero() {
+		log.Warnf("Prematurely received 'init' from %v at step %v", makerTaker(stepInfo.actor.isMaker), stepInfo.step)
+	} else if deadline := lastEvent.Add(s.bTimeout); expireTime.After(deadline) {
+		expireTime = deadline
+	}
+	log.Debugf("Allowing until %v (%v) to locate contract from %v (%v), match %v",
+		expireTime, time.Until(expireTime), makerTaker(stepInfo.actor.isMaker),
+		stepInfo.step, matchID)
+
```

Comment:
This is a fair bit of complexity for little gain in appropriateness of the latencyQ waiter timeout.  But it is more appropriate than always going with txWaitExpiration from *now*.